### DISCUSSION TITLE
PyMySQL Operational Error 1045, ACCESS DENIED (PASSWORD: YES) when MYSQL_ROOT_PASSWORD is supplied.

### DISCUSSION THREAD

**Discussion by Nsandomeno** [PyMySQL/PyMySQL#909] [Q&A]
**Describe the bug**
PyMySQL fails to connect to MySQL Docker Container. Inside the MySQL Docker Container, the logs produce an `mbind: Operation not permitted.` message while the backend repeatedly attempts to connect to the MySQL container but produces the error message: `ERROR:__main__:(pymysql.err.OperationalError) (1045, "Access denied for user 'root'@'172.18.0.8' (using password: YES)")`

**To Reproduce**
- Build a Python backend and MySQL docker container with docker-compose. (both build successfully) database logs after the build:

![database_logs_after_docker-compose](https://user-images.githubusercontent.com/49354057/102639051-76a4b300-4126-11eb-8452-955d2fb3d6c3.png)

- Run a script that initializes the backend connection to the MySQL database container... the connection url scheme that is used is `mysql+pymysql` and the dependency pymysql is being brought in with the extra dependency: `pymysql[rsa]`. 

The connection url is completed with: `mysql+pymysql://root:the correct password I promise@db/app`

Where `db` is the name of the service in the docker-compose.yml file that builds the MySQL (I get a connection refused error if I use 0.0.0.0 127.0.0.1 or localhost:<port>) and `app` is the name of the databased created, validated by the logs.

![Access-denied](https://user-images.githubusercontent.com/49354057/102639983-d3ed3400-4127-11eb-8243-a14177af775d.png)

- Revisiting the logs after this it has produced an `mbind: Operation not permitted` message.

The `db` docker-compose.yml service code: 

![docker-compose](https://user-images.githubusercontent.com/49354057/102640138-072fc300-4128-11eb-8417-a0ff1f1e6795.png)

The `db` docker-compose.override (for some local dev additions):

![docker-compose override](https://user-images.githubusercontent.com/49354057/102640279-3b0ae880-4128-11eb-8fb4-77f54848923e.png)

**Expected behavior**
A connection.

**Environment**
 - OS: MacOS up-to-date
 - Server and version: Docker version 19.03.8, MySQL Docker Image (Official): https://hub.docker.com/_/mysql
 - PyMySQL version: 0.10.1

**Comment by Nsandomeno** [comment]
What disqualifies this as an issue? @methane

**Reply by methane** [reply]
If you want to report "issue" (e.g. bug in this library), you shouldn't describe *your environment* and messages you seen in *your environemnt*. Because issue report is not "Help me! Please fix my trouble!".

You should reproduce it with minimum environment (e.g. Setup MySQL with Docker, no config file, no SQLAlchemy), and share complete step to reproduce instead (using code block, instead of screenshot). And you can report what you expect and what you seen in the minimum reproducing environment.

**Reply by Nsandomeno** [reply]
Understood, thanks. I will fix this up in the next few days. I have reproduced in a more vanilla environment but still using SQLAlchemy so I will get that out next.
